Sources/AsyncHTTPClient/ConnectionPool/State Machine/HTTPConnectionPool+HTTP1Connections.swift

```diff
@@ -532,12 +532,18 @@ extension HTTPConnectionPool {
 
         mutating func migrateToHTTP2() -> HTTP1ToHTTP2MigrationContext {
             var migrationContext = HTTP1ToHTTP2MigrationContext()
-            self.connections.removeAll { connection in
-                switch connection.migrateToHTTP2(&migrationContext) {
+            let initialOverflowIndex = self.overflowIndex
+
+            self.connections = self.connections.enumerated().compactMap { index, connectionState in
+                switch connectionState.migrateToHTTP2(&migrationContext) {
                 case .removeConnection:
-                    return true
+                    if index < initialOverflowIndex {
+                        self.overflowIndex = self.connections.index(before: self.overflowIndex)
```

Comment:
Previously we didn't update `self.overflowIndex` correctly which did hide some bugs because we rarely reused `HTTP1Connections` ever again after migration.